tests/components/backup/test_http.py

```diff
@@ -59,6 +63,129 @@ async def test_downloading_remote_backup(
         assert await resp.content.read() == b"backup data"
 
 
+async def test_downloading_local_encrypted_backup_file_not_found(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+) -> None:
+    """Test downloading a local backup file."""
+    await setup_backup_integration(hass)
+    client = await hass_client()
+
+    with patch(
+        "homeassistant.components.backup.backup.CoreLocalBackupAgent.async_get_backup",
+        return_value=TEST_BACKUP_ABC123,
+    ):
+        resp = await client.get(
+            "/api/backup/download/abc123?agent_id=backup.local&password=blah"
+        )
+        assert resp.status == 404
+
+
+@pytest.mark.usefixtures("mock_backups")
+async def test_downloading_local_encrypted_backup(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+) -> None:
+    """Test downloading a local backup file."""
+    await setup_backup_integration(hass)
+    await _test_downloading_encrypted_backup(hass_client, "backup.local")
+
+
+async def aiter_from_iter(iterable: Iterable) -> AsyncIterator:
+    """Convert an iterable to an async iterator."""
+    for i in iterable:
+        yield i
+
+
+@patch.object(BackupAgentTest, "async_download_backup")
+async def test_downloading_remote_encrypted_backup(
+    download_mock,
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+) -> None:
+    """Test downloading a local backup file."""
+    backup_path = get_fixture_path("test_backups/ed1608a9.tar", DOMAIN)
+    await setup_backup_integration(hass)
+    hass.data[DATA_MANAGER].backup_agents["domain.test"] = BackupAgentTest(
```

Comment:
Mock a platform with this agent instead of adding the agent directly to the manager.

Reply:
This pattern is used in tests already, let's improve it in a separate PR 👍

Reply:
Mocking a platform is also used in tests already.

Reply:
We have this helper:

https://github.com/home-assistant/core/blob/5580a78a4da6d79b1e42ded14c69d0022f9b0b16/tests/components/backup/common.py#L171

Reply:
It turns out the `setup_backup_integration` also can set up test agents, so I used that.